### PR TITLE
Correct drift using PID loop

### PR DESCRIPTION
**Pull request by jherrmann06**

### CODE REVIEW

File: src/main/java/frc/robot/util/JoystickModification.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public static double modifyAxis(double value) {
+    value = deadband(value, 0.05);
+
+    value = Math.copySign(value * value, value);
+
+    double scaler = speedScaler.getDouble(1.0);
+
+    value = value * scaler; //scale down value to slow robot
+
+    return value;
```
Comment (jdlanning) [inline src/main/java/frc/robot/util/JoystickModification.java:68]: Formatting / indentation looks a little odd in this method. Suggest to auto format the entire file

File: src/main/java/frc/robot/commands/DefaultDriveCommand.java [hunk 2]
```diff
@@ -27,15 +31,26 @@ public DefaultDriveCommand(DrivetrainSubsystem drivetrainSubsystem,
...
-        // You can use `new ChassisSpeeds(...)` for robot-oriented movement instead of field-oriented movement
-        m_drivetrainSubsystem.drive(
-                ChassisSpeeds.fromFieldRelativeSpeeds(
-                        m_translationXSupplier.getAsDouble(),
-                        m_translationYSupplier.getAsDouble(),
-                        m_rotationSupplier.getAsDouble(),
-                        m_drivetrainSubsystem.getGyroscopeRotation()
-                )
-        );
+        if (m_rotationSupplier.getAsDouble() == 0 && (m_translationXSupplier.getAsDouble() != 0 || m_translationYSupplier.getAsDouble() != 0)) {
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/DefaultDriveCommand.java]: I think this if can be simplified to just check the rotation. It’s ok for the pid to remain on when no joysticks are moving

File: src/main/java/frc/robot/commands/DefaultDriveCommand.java [hunk 3]
```diff
@@ -27,15 +31,26 @@ public DefaultDriveCommand(DrivetrainSubsystem drivetrainSubsystem,
...
+            if (updateTarget) {
+                target = m_drivetrainSubsystem.getGyroscopeRotation();
+                updateTarget = false;
+            }
+            m_drivetrainSubsystem.drive(
+                    ChassisSpeeds.fromFieldRelativeSpeeds(
+                            m_translationXSupplier.getAsDouble(),
+                            m_translationYSupplier.getAsDouble(),
+                            -pid.calculate(m_drivetrainSubsystem.getGyroscopeRotation().minus(target).getRadians()),
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/DefaultDriveCommand.java]: This code would read a little better if you wrapped the conditional logic inside another method called getRotation() then your execute would be pretty similar to before

File: src/main/java/frc/robot/commands/DefaultDriveCommand.java [hunk 4]
```diff
@@ -1,22 +1,26 @@
...
 
     private final DoubleSupplier m_translationXSupplier;
     private final DoubleSupplier m_translationYSupplier;
     private final DoubleSupplier m_rotationSupplier;
 
-    public DefaultDriveCommand(DrivetrainSubsystem drivetrainSubsystem,
-                               DoubleSupplier translationXSupplier,
-                               DoubleSupplier translationYSupplier,
-                               DoubleSupplier rotationSupplier) {
+    private final PIDController pid = new PIDController(3, 0.5, 0.5);
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/DefaultDriveCommand.java]: Suggest to move to constants

File: src/main/java/frc/robot/util/JoystickModification.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
+package frc.robot.util;
+
+import java.util.Map;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+
+public class JoystickModification {
```
Comment (jdlanning) [inline src/main/java/frc/robot/util/JoystickModification.java:10]: This is a nice encapsulation of related code from RobotContainer. Helps to keep that other class focused on robot integration / composition and this new class for joystick considerations

**Review by jdlanning — COMMENTED** [review]
Also glad to see this come together . It feels like we are getting close to having many of our telep drive basics complete

**Comment by jherrmann06** [comment]
Responded to feedback

**Review by jdlanning — APPROVED** [review]

**Pull request merged**